### ISSUE TITLE
MD026 should not fire for semi-colons in a html entity

### ISSUE THREAD

**Issue by jeroenheijmans** [DavidAnson/markdownlint#62]
Please consider this snippet of markdown:

```md
# Java vs C&#35;
```

The VSCode extension "markdownlint" version 0.9.1 will present a **MD026** warning like this:

![image](https://user-images.githubusercontent.com/1590536/27070383-13e9fe02-5019-11e7-9546-ab57d8317e7b.png)

So it shows:

> Click for more information about MD026

and highlights the semi-colon.

If possible the rule should not trigger on html entities. If this isn't feasable it would be nice to have a footnote in [linked docs](https://github.com/DavidAnson/markdownlint/blob/v0.5.0/doc/Rules.md#md026) explaining a workaround (if any).

**Comment by DavidAnson** [comment]
The documentation does mention that the `punctuation` parameter can be used to change the list of characters: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md026

I'll think about special treatment for HTML entities. Thanks!

**Comment by DavidAnson** [comment]
Thinking about this more, there may be a better way. I think you're probably trying to do this:

```md
# Java vs C#
```

But that triggers `MD020/no-missing-space-closed-atx` because of the hash at the end. So you escaped with an HTML entity.

Instead, what if you escaped via the Markdown convention of '\\'? That would look like this:

```md
# Java vs C\#
```

And triggers no warnings from `markdownlint` and has the benefit of being native Markdown (i.e., no assumption about rendering to HTML).

I think this is a better solution and will close the issue if you agree.

Read more here if interested: https://daringfireball.net/projects/markdown/syntax#backslash

**Comment by jeroenheijmans** [comment]
Your analysis is spot on. I was indeed trying to escape the `#` but failed to realize I could've done so with a `\` instead of a html entity.

Thanks for your time, insights, and in general a great plugin 👍
